com.unity.ml-agents/Runtime/Sensors/StackingSensor.cs

```diff
@@ -128,17 +146,88 @@ public string GetName()
         }
 
         /// <inheritdoc/>
-        public virtual byte[] GetCompressedObservation()
+        public byte[] GetCompressedObservation()
+        {
+            var compressed = m_WrappedSensor.GetCompressedObservation();
+            m_StackedCompressedObservations[m_CurrentIndex] = compressed;
+
+            int bytesLength = 0;
+            foreach (byte[] compressedObs in m_StackedCompressedObservations)
+            {
+                bytesLength += compressedObs.Length;
+            }
+
+            byte[] outputBytes = new byte[bytesLength];
+            int offset = 0;
+            for (var i = 0; i < m_NumStackedObservations; i++)
+            {
+                var obsIndex = (m_CurrentIndex + 1 + i) % m_NumStackedObservations;
+                Buffer.BlockCopy(m_StackedCompressedObservations[obsIndex],
+                    0, outputBytes, offset, m_StackedCompressedObservations[obsIndex].Length);
+                offset += m_StackedCompressedObservations[obsIndex].Length;
+            }
+
+            return outputBytes;
+        }
+
+        public int[] GetCompressedChannelMapping()
         {
-            return null;
+            return m_CompressionMapping;
         }
 
         /// <inheritdoc/>
-        public virtual SensorCompressionType GetCompressionType()
+        public SensorCompressionType GetCompressionType()
         {
-            return SensorCompressionType.None;
+            return m_WrappedSensor.GetCompressionType();
         }
 
-        // TODO support stacked compressed observations (byte stream)
+        /// <summary>
+        /// Create Empty PNG for initializing the buffer for stacking.
+        /// </summary>
+        public byte[] CreateEmptyPNG()
+        {
+            int height = m_WrappedSensor.GetObservationShape()[0];
+            int width = m_WrappedSensor.GetObservationShape()[1];
+            var texture2D = new Texture2D(width, height, TextureFormat.RGB24, false);
+            return texture2D.EncodeToPNG();
+        }
+
+        public int[] ConstructStackedCompressedChannelMapping(ISensor wrappedSenesor)
+        {
+            int[] wrappedMapping;
+            int wrappedNumChannel = wrappedSenesor.GetObservationShape()[2];
+            var compressibleSensor = m_WrappedSensor as ISparseChannelSensor;
+            if (compressibleSensor != null)
+            {
+                wrappedMapping = compressibleSensor.GetCompressedChannelMapping();
```

Comment:
Do you need to handle GetCompressedChannelMapping returning null?

I think you could do something like
```
if (compressibleSensor != null)
{
    wrappedMapping = compressibleSensor.GetCompressedChannelMapping();
}
if(wrappedMapping = null)
{
    if (wrappedNumChannel == 1) {...}
    else {}
}
```